### ISSUE TITLE
Generic types referencing @mui/x-date-pickers wrong after v2.13.0 

### ISSUE THREAD

**Issue by lpicchi** [KevinVandy/material-react-table#1320]
### material-react-table version

v2.13.0 

### react & react-dom versions

v18.3.1

### Describe the bug and the steps to reproduce it

When using version > v2.13.0 some types definitions referencing @mui/x-date-pickers are wrongly using generics

Example:

- Before v2.13.0 (Correct use of generics):

![image](https://github.com/user-attachments/assets/a62352e1-1777-49d8-bed1-864a9841c076)

- After v2.13.0 (Incorrect use of generics):

![image](https://github.com/user-attachments/assets/844013be-8c2f-4f78-bc9b-e4ed7a860a4a)
![image](https://github.com/user-attachments/assets/14c3d7e9-99f8-41a9-b5f5-dd97ac54a4cc)

Looking at [`src/types.ts`](https://github.com/KevinVandy/material-react-table/blob/v2/packages/material-react-table/src/types.ts) for v2 it looks fine. But the types generated in `dist` when building seems to be different and wrong

For v2.13.0 there are six places where types are wrong

![image](https://github.com/user-attachments/assets/d511938d-70ec-4968-bbc2-83c3820597c5)

But for v2.13.1 only 3 places

![image](https://github.com/user-attachments/assets/73700536-a6f1-4402-8995-f03b133c73fb)

So it seems part of the issue has been solved between v2.13.0 and v2.13.1 somehow

I'm not able to use tools like `dts-bundle-generator` because of this problem and had to downgrade to `2.12.1`

### Minimal, Reproducible Example - (Optional, but Recommended)

Optional

### Do you intend to try to help solve this bug with your own PR?

Maybe, I'll investigate and start debugging

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by idesigncode** [comment]
I've recently found this error when updating my version of `@mui/x-date-pickers` to v8.5.2 - there was a breaking change in v8 to the `...PickerProps<never>` types that [removed passing the date object as a generic](https://mui.com/x/migration/migration-pickers-v7/#do-not-pass-the-date-object-as-a-generic).

Initially I tried an override but these don't work on package peer-dependencies:
```json
    "overrides": {
        "material-react-table": {
            "@mui/x-date-pickers": "^7.29.2"
        }
    }
```

My current (hopefully temporary) solution has been to add a local patch that changes the types `...PickerProps<never>` to `...PickerProps` in `/node_modules/material-react-table/dist/index.d.ts` but a proper fix would be to address this in the source declaration file (perhaps conditionally giving the generic if needed?).

TLDR: The `@mui/x-date-pickers` peer-dependency version is not restricted to the major version that it needs: e.g. `>= 7.15` (should be `^7.15`)
